Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js

```diff
@@ -642,3 +737,211 @@ describe('gdjs.PlatformerObjectRuntimeBehavior and gdjs.PlatformRuntimeBehavior
     expect(object.getY()).to.be(-30);
   });
 });
+
+describe('ladder', function() {
+  var runtimeScene;
+  var object;
+  var scale;
+
+  beforeEach(function() {
+    runtimeScene = makeTestRuntimeScene();
+
+    // Put a platformer object on a platform
+    object = new gdjs.RuntimeObject(runtimeScene, {
+      name: 'obj1',
+      type: '',
+      behaviors: [
+        {
+          type: 'PlatformBehavior::PlatformerObjectBehavior',
+          name: 'auto1',
+          gravity: 1500,
+          maxFallingSpeed: 1500,
+          acceleration: 500,
+          deceleration: 1500,
+          maxSpeed: 500,
+          jumpSpeed: 900,
+          canGrabPlatforms: true,
+          ignoreDefaultControls: true,
+          slopeMaxAngle: 60,
+          jumpSustainTime: 0.2,
+          roundCoordinates: true,
+        },
+      ],
+    });
+    object.getWidth = function() {
+      return 10;
+    };
+    object.getHeight = function() {
+      return 20;
+    };
+    runtimeScene.addObject(object);
+
+    // Put a platform.
+    const platform = addPlatformObject(runtimeScene);
+    platform.setPosition(0, -10);
+
+    ladder = addLadderObject(runtimeScene);
+    ladder.setPosition(30, -10 - ladder.getHeight());
+  });
+
+  it('can climb a ladder', function() {
+    object.setPosition(30, -32);
+    // Ensure the object falls on the platform
+    for (var i = 0; i < 10; ++i) {
+      runtimeScene.renderAndStep(1000 / 60);
+    }
+
+    //Check the object is on the platform
+    expect(object.getY()).to.be(-30); // -30 = -10 (platform y) + -20 (object height)
+    expect(object.getBehavior('auto1').isFalling()).to.be(false);
+    expect(object.getBehavior('auto1').isMoving()).to.be(false);
+
+    // Climb the ladder
+    object.getBehavior('auto1').simulateLadderKey();
+    runtimeScene.renderAndStep(1000 / 60);
+    for (var i = 0; i < 10; ++i) {
+      object.getBehavior('auto1').simulateUpKey();
+      runtimeScene.renderAndStep(1000 / 60);
+    }
+    expect(object.getY()).to.be.below(-30);
+
+    // Stay on the ladder
+    const stayY = object.getY();
+    for (var i = 0; i < 10; ++i) {
+      runtimeScene.renderAndStep(1000 / 60);
+    }
+    expect(object.getY()).to.be(stayY);
+    expect(object.getBehavior('auto1').isOnLadder()).to.be(true);
+
+    // Climb the ladder
+    for (var i = 0; i < 14; ++i) {
+      object.getBehavior('auto1').simulateUpKey();
+      runtimeScene.renderAndStep(1000 / 60);
+    }
+    // Check that we reached the maximum height
+    const playerAtLadderTop = ladder.getY() - object.getHeight();
+    expect(object.getY()).to.be.within(playerAtLadderTop, playerAtLadderTop + 2);
+
+    //TODO it fall at the end but that may not be expected.
```

Comment:
It's kind of expected because when you're on the top of a ladder, it's not considered as a platform so you start to fall. Feel free to add a check, checking this is the case for one frame.

Reply:
Thinking more about it, it's easier to add events to make the player stop at the top of the ladder that the other way around. So, it makes sense.

Reply:
Yep, though we could also add an option later to avoid falling when on top (like a jump thru), or creators can add a jump thru, etc... Easier this way than mandating a stop at the end :)